src/controllers/auth.controller.ts

```diff
@@ -67,4 +68,54 @@ export const getHospitalDetail = async (req: Request, res: Response) => {
     } catch (error) {
         return res.status(500).json({ success: false, message: 'Error occurred when fetching data!' })
     }
+}
+
+export const registerPatient = async (req: Request, res: Response) => {
+    let data: patientIn = req.body;
+
+    try {
+        const error = checkValidity(data, patientRegRule);
+        if(error) {
+            return res.status(400).json({
+                success: false, message: error
+            })
+        }
+        data.password = await hashPassword(data.password);
+
+        let result: patientOut = await registerAPatient(req.body)
+        return res.status(201).json({ success: true, data: result });
+
+    } catch (error: any) {
+        return res.status(412).json({ success: false, message: error.message });
+    }
+};
+
+export const loginPatient = async(req: Request, res: Response) => {
```

Comment:
A login route is not required for patient as they won't be logging in on this platform.